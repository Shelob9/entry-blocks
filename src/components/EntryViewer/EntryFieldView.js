import React, {Fragment} from 'react';

export const EntryFieldView = (props) => {
	return <Fragment>{props.entryField.value}</Fragment>
};

