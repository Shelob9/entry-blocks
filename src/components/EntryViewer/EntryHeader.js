import React, {Fragment} from 'react';
export const EntryHeader = (props ) => {
	const {entryField,labelWith} = props;
	const label = labelWith && labelWith.length ? labelWith : entryField.slug;
	return (
		<Fragment>
			{label}
		</Fragment>

	);
};
