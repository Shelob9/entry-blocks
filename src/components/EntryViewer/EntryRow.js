import React from 'react';
import {EntryFieldView} from "./EntryFieldView";
import {TableCell} from "./components/TableCell";

export const EntryRow = (props ) => {
	const {entry} = props;
	const {
		id,
		fields
	} = entry;

	return (
		<tr>
			<th scope="row">{id}</th>
			{Object.values(fields).map( (entryField) => {
				return (
					<TableCell key={entryField.id}>
						<EntryFieldView entryField={entryField} labelWith={entryField.slug} fieldType={'text'}/>
					</TableCell>
				);
			})}
		</tr>
	);
};
