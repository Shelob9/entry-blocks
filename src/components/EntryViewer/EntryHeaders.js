
import React from 'react';
import {EntryHeader} from "./EntryHeader";

export function getFieldFromCollection( fields, fieldId )  {
	if( ! fields.hasOwnProperty(fieldId) ){
		// eslint-disable-next-line no-throw-literal
		throw 'Can not find field';
	}
	return fields[fieldId];
};


export const EntryHeaders = (props) => {
	const {formFields} = props;
	const firstEntry = Object.values(props.entries)[0];
	return (
		<tr>
			<th scope="col">ID</th>
			{Object.values(firstEntry.fields).map( (entryField ) => {
				let labelWith;
				let fieldType = 'text';

				try {
					const field = getFieldFromCollection(formFields,entryField.field_id);
					labelWith = field.name;
					fieldType = field.type;
				}catch(e){
					labelWith = entryField.slug
				}


				return (
					<th key={entryField.id}>
						<EntryHeader entryField={entryField} labelWith={labelWith} fieldType={fieldType}/>
					</th>);
			})}
		</tr>
	);
};
