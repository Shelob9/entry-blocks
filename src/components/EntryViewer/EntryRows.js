import React, {Fragment} from 'react';
import {EntryRow} from "./EntryRow";
export const EntryRows = (props: {entries} ) => {
	return (
		<Fragment>
			{Object.values(props.entries).map( (entry : Entry ) => {
				return (<EntryRow
					entry={entry}
					key={entry.id}
				/>)
			})}
		</Fragment>
	);
}
