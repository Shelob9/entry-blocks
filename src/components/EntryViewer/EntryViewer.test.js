import renderer from "react-test-renderer";
import React from "react";

import Enzyme, {mount,shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {EntryFieldView} from "./EntryFieldView";
import {EntryRow} from "./EntryRow";
import {EntryRows} from "./EntryRows";
import {EntryHeader} from "./EntryHeader";
import {EntryHeaders} from "./EntryHeaders";


Enzyme.configure({adapter: new Adapter()});



const _forms = require( '../../__MOCK_DATA__/forms' );
const forms = Object.values(_forms );
const formId = 'CF5bde22b646ed0';
const form = _forms[formId];
const formFields = form.fields;

const entries = require( '../../__MOCK_DATA__/entries' );
const entry = entries[0];
const entryFieldId = 'fld_8768091';
const entryId = entry.id;
const entryField = entry.fields[entryFieldId];

const instanceId = '54fds';
const beforeText = 'Before';
const afterText = 'After';
const onSetForm = jest.fn();
const onSetEntry = () => jest.fn();
const onSetEntryField =jest.fn();
const setBefore = () => jest.fn();
const setAfter = () => jest.fn();




describe( 'Entry Viewer components ', () => {
	test( 'EntryFieldView matches snapshot', () => {
		expect(
			renderer.create(
				<EntryFieldView
					entryField={entryField}
					fieldType={'text'}
				/>
			)
		).toMatchSnapshot();
	});

	test( 'EntryHeader matches snapshot with labelWith', () => {
		expect(
			renderer.create(
				<EntryHeader
					entryField={entryField}
					labelWith={'Label Me!'}
				/>
			)
		).toMatchSnapshot();
	});


	test( 'EntryHeader matches snapshot without label', () => {
		expect(
			renderer.create(
				<EntryHeader
					entryField={entryField}
				/>
			)
		).toMatchSnapshot();
	});


	test( 'EntryHeaders matches snapshot ', () => {
		expect(
			renderer.create(
				<EntryHeaders
					entries={entries}
					formFields={formFields}
				/>
			)
		).toMatchSnapshot();
	});



	test( 'EntryFieldView matches snapshot', () => {
		expect(
			renderer.create(
				<EntryFieldView
					entryField={entryField}
					fieldType={'text'}
				/>
			)
		).toMatchSnapshot();
	});


	test( 'EntryRow matches snapshot', () => {
		expect(
			renderer.create(
				<EntryRow entry={entry}/>
			)
		).toMatchSnapshot();
	});


	test( 'EntryRow matches snapshot', () => {
		expect(
			renderer.create(
				<EntryRows
					entries={entries}
				/>
			)
		).toMatchSnapshot();
	});
});