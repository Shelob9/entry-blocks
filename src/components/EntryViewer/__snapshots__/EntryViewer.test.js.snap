exports[`Entry Viewer components  EntryFieldView matches snapshot 1`] = `"Josh"`;

exports[`Entry Viewer components  EntryFieldView matches snapshot 2`] = `"Josh"`;

exports[`Entry Viewer components  EntryHeader matches snapshot with labelWith 1`] = `"Label Me!"`;

exports[`Entry Viewer components  EntryHeader matches snapshot without label 1`] = `"first_name"`;

exports[`Entry Viewer components  EntryHeaders matches snapshot  1`] = `
<tr>
  <th
    scope="col">
    ID
  </th>
  <th>
    first_name
  </th>
  <th>
    last_name
  </th>
  <th>
    email_address
  </th>
  <th>
    comments_questions
  </th>
</tr>
`;

exports[`Entry Viewer components  EntryRow matches snapshot 1`] = `
<tr>
  <th
    scope="row">
    1
  </th>
  <td>
    Josh
  </td>
  <td>
    Pollock
  </td>
  <td>
    jpollock412@gmail.com
  </td>
  <td>
    Message
    
    Second Line
  </td>
</tr>
`;

exports[`Entry Viewer components  EntryRow matches snapshot 2`] = `
Array [
  <tr>
    <th
        scope="row">
        1
    </th>
    <td>
        Josh
    </td>
    <td>
        Pollock
    </td>
    <td>
        jpollock412@gmail.com
    </td>
    <td>
        Message
        
        Second Line
    </td>
</tr>,
  <tr>
    <th
        scope="row">
        2
    </th>
    <td>
        Roy
    </td>
    <td>
        Sivan
    </td>
    <td>
        roy@hiroy.club
    </td>
    <td>
        fs
    </td>
</tr>,
]
`;
